integrations/LaunchDarkly/browser.js

```diff
@@ -0,0 +1,84 @@
+import {
+    isArray,
+    isObject,
+} from "../../utils/utils"; import logger from "../../utils/logUtil";
+import ScriptLoader from "../ScriptLoader";
+const isString = function isString(val) {
+    return toStr$1.call(val) === "[object String]";
+};
+class LaunchDarkly {
+    constructor(config) {
+        this.name = "LaunchDarkly";
+        this.clientSideId = config.clientSideId;
+    }
+    init() {
+        logger.debug("===in init LaunchDarkly===");
+        if (!this.clientSideId) {
+            logger.debug("clientSideId is missing");
+            return;
+        }
+        ScriptLoader(null, "https://unpkg.com/launchdarkly-js-client-sdk@2");
+    }
+    isLoaded() {
+        logger.debug("===In isLoaded LaunchDarkly===");
+        return !!window.LDClient;
+    }
+    isReady() {
+        logger.debug("===In isReady LaunchDarkly===");
+        if (window.LDClient)
+            return true;
+        else
+            return false;
+    }
+    identify(rudderElement) {
+        var user = {};
+        const { message } = rudderElement;
+        user.key = message.userId || message.anonymousId;
+        const { traits } = message.context;
+        if (traits.anonymous !== undefined)
+            user.anonymous = traits.anonymous;
+        if (traits.avatar !== undefined)
+            user.avatar = traits.avatar
+        if (traits.country != undefined)
+            user.country = traits.country
+        if (traits.custom != undefined && isObject(traits.custom))
+            user.custom = traits.custom;
+        if (traits.email != undefined)
+            user.email = traits.email;
+        if (traits.ip != undefined && isString(traits.ip))
+            user.ip = traits.ip;
+        if (traits.lastName != undefined && isString(traits.lastName))
+            user.lastName = traits.lastName;
+        if (traits.name != undefined && isString(traits.name))
+            user.name = traits.name;
+        if (traits.privateAttributeNames != undefined && isArray(traits.privateAttributeNames))
+            user.privateAttributeNames = traits.privateAttributeNames;
+        if (traits.secondary != undefined && isString(traits.secondary))
+            user.secondary = traits.secondary;
+        if (window.ldclient) {
```

Comment:
`ldclient` and `LDClient` are two different things?

Reply:
Yes, they are different. While creating a user I am using LDClient to initialize ldclient object. While updating I am updating the existing user with ldclient.identify.

Reply:
Two things to be fixed here:
- Use `window.LDClient` instead of `LDClient`.
- Don't use `window.ldclient` instead declare a class-level variable.

Reply:
okay.

Reply:
You can store it at the class level

Reply:
The problem with that implementation is, the customer may need to use the ldclient object which will help him to use the feature flag. ldclient object has a function named variation which helps to get status of a feature-flag. So if I declare it as class-level variable that will not be accessible by the customer.